roofit/roofitcore/src/FitHelpers.cxx

```diff
@@ -83,18 +96,47 @@ int calcAsymptoticCorrectedCovariance(RooAbsReal &pdf, RooMinimizer &minimizer,
          num(k, l) = 0.0;
       }
    }
-   RooArgSet obs;
-   pdf.getObservables(data.get(), obs);
+
    // Create derivative objects
    std::vector<std::unique_ptr<RooDerivative>> derivatives;
    const RooArgList &floated = rw->floatParsFinal();
    std::unique_ptr<RooArgSet> floatingparams{
-      static_cast<RooArgSet *>(pdf.getParameters(data)->selectByAttrib("Constant", false))};
+     static_cast<RooArgSet *>(logpdf.getParameters(data)->selectByAttrib("Constant", false))};
```

Comment:
```suggestion
   RooArgSet allparams;
   logpdf.getParameters(*data.get(), allparams);
   std::unique_ptr<RooArgSet> floatingparams{static_cast<RooArgSet *>(allparams.selectByAttrib("Constant", false))};
```
To fix the memory leak from not deleting the return value of `getParameters()`.

Reply:
thanks